### ISSUE TITLE
Drift dev warning on BaseRelationTable

### ISSUE THREAD

**Issue by ebelevics** [simolus3/moor#1955]
This is class that I use to extend on other many-to-many tables in order to write join methods inside BaseDao

```dart
abstract class BaseRelationTable<P extends Table, C extends Table> extends Table {
  IntColumn get parentId => integer().references(P, #localId, onDelete: KeyAction.cascade)();
  IntColumn get childId => integer().references(C, #localId, onDelete: KeyAction.cascade)();

  @override
  Set<Column> get primaryKey => {parentId, childId};
}
```

Everything works fine, but the console always prints:

```
There were some errors while running moor_generator on package:***.dart:
[INFO] drift_dev:drift_dev on ***..dart:line 4, column 50 of ***..dart: P is not a class!
  ╷
4 │   IntColumn get parentId => integer().references(P, #localId, onDelete: KeyAction.cascade)();

```

Is there a way to remove this warning?

**Comment by simolus3** [comment]
Wait, does this really work? Do instantiations of `BaseRelationTable` get generated with the correct foreign key? Drift shouldn't analyze the `BaseRelationTable` in isolation, but it looks like this also breaks for the actual table classes inheriting from it. I think it's pretty neat and drift should support it, but judging from the error it sounds like the reference isn't generated correctly.

**Comment by ebelevics** [comment]
Yes, after some while I noticed that Delete cascade didn't work, and by inspecting generated file, it just wasn't there on defaultConstraint. So I came with this idea, not as clean as previous, but it does the job.

```dart
class RelationTable extends BaseRelationTable {
  RelationTable() {
    init(firstId, secondId);
  }

  IntColumn get firstId => integer().references(TableOne, #localId, onDelete: KeyAction.cascade)();
  IntColumn get secondId => integer().references(TableTwo, #localId, onDelete: KeyAction.cascade)();

  @override
  Set<Column> get primaryKey => {firstId, secondId};
}

abstract class BaseRelationTable extends Table {
  late final Column<int?> parentId;
  late final Column<int?> childId;

  init(Column<int?> parentColumn, Column<int?> childColumn) {
    this.parentId = parentColumn;
    this.childId = childColumn;
  }
}
```

and then I can do something like this in baseDao:
```dart
  Future<List<D>> getByRelation_(dynamic parentId, TableInfo<BaseRelationTable, DataClass> relTable) async {
    final contentQuery = select(relTable).join(
      [innerJoin(_table, _table.asDslTable.localId.equalsExp(relTable.asDslTable.childId))],
    )..where(relTable.asDslTable.parentId.equals(parentId));
    final rows = await contentQuery.get();

    return rows.map((r) => r.readTable(_table)).toList();
  }
```

**Comment by simolus3** [comment]
I think you could also do:

```dart
abstract class BaseRelationTable extends Table {
  Column<int?> get parentId;
  Column<int?> get childId;

  @override
  Set<Column> get primaryKey => {parentId, childId};
}

class RelationTable extends BaseRelationTable {
  @override
  IntColumn get parentId => integer().references(TableOne, #localId, onDelete: KeyAction.cascade)();
  @override
  IntColumn get childId => integer().references(TableTwo, #localId, onDelete: KeyAction.cascade)();
}
```

(Unless it is your intention to use different column names, in which case your existing solution is a good way to go)

**Comment by ZhulanovAA** [comment]
@simolus3, I've actualy just made almost the same thing as your last example:


``` Dart
abstract class DConnectionToExportPackTable extends Table {
  IntColumn get exportPackId => integer().references(DExportPacks, #id)();
  IntColumn get dataPackId;

  @override
  Set<Column>? get primaryKey => {exportPackId, dataPackId};
}
```

It seems that all inhereting tables are generated properly without any errors, but generator still gives warnings for the base class, even though it's abstract:

```
[WARNING] drift_dev on lib/data/drift/tables/shared.dart:
*my drift file*: This getter does not create a valid column that can be parsed by drift. Please refer to the readme from drift to see how columns are formed. If you have any questions, feel free to raise an issue.
   ╷
81 │   IntColumn get dataPackId;
   │                 ^^^^^^^^^^
   ╵
[WARNING] drift_dev on lib/data/drift/tables/shared.dart:
*my drift file*: Column not found in this table
   ╷
84 │   Set<Column>? get primaryKey => {exportPackId, dataPackId};
   │                                                 ^^^^^^^^^^
   ╵

```

Is it expected? Shuld I open a separate issue for that?

**Comment by simolus3** [comment]
I think this is a regression in drift 2.3. In our new analyzer, we always analyze all tables instead of just the ones explicitly added to the database. This enables modular analysis and is generally faster, but it now means that we consider `DConnectionToExportPackTable` to be an actual table and complain about it.

I think I will just make the analyzer ignore table classes that have abstract methods.

**Comment by simolus3** [comment]
Should be fixed again in bbb19d59bb1eeb6c3e1c96329ea4771ebb75fee0 - thanks for reporting the problem.
